content/tutorials/MQTT-snake-js.textile

```diff
@@ -0,0 +1,267 @@
+
+For authentication, you'll need to either use an API Key, or a token. For simplicity of the tutorial, you'll be using the API key you obtained in step 1. For this, you'll need to provide the first half of the API Key (before the colon) as the username, and the second half of the API Key (after the colon) as the password. For example, if your API key is @A12B3C.4D:5E6F7G8H@, your username will be @A12B3C.4D@ and your password will be @5E6F7G8H@.
+
+Add the following code below your current code in @controller.js@, replacing 'FIRST_HALF_OF_API_KEY' and 'SECOND_HALF_OF_API_KEY' with the appropriate parts of your own API Key:
+
+```[javascript]
+  var options = { keepAlive: 15,
+    username: 'FIRST_HALF_OF_API_KEY',
+    password: 'SECOND_HALF_OF_API_KEY',
+    port: 8883
+  };
+
+  var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+
+  client.on('connect', function () {
+    console.log('connected!');
+  });
+
+  client.on('error', function(err){
+    console.log(err);
+    client.end();
+  });
+```
+
+With the above code, you set @options@ as described above, and then attempt to connect to Ably's MQTT endpoint. You specify @mqtts@ as part of the host address to specify that the connection attempt should be made with MQTT using SSL/TLS.
+
+Now, try out the code by typing @node controller.js@ in the command line and you should see in the console that the client successfully connects through MQTT.
+
+<a href="/images/tutorials/mqtt-snake-connected.png" target="_blank">
+  <img src="/images/tutorials/mqtt-snake-connected.png" style="width: 100%" alt="Connected through MQTT!">
+</a>
+
+If you have any issues with this, please check out our "usage notes":https://support.ably.io/solution/articles/3000074906-using-the-mqtt-protocol-adapter.
+
+"See this step in Github":https://github.com/ably/tutorials/commit/ec7106af325d4c65b7f7bd3f3183e76df89c5529
+
+h2. Step 6 - Send key presses through MQTT
+
+Now that you have established a connection through MQTT to Ably, you can send some data through it. You need to firstly create a function to easily publish messages for the key presses you're interested in. Add the following function to the bottom of @controller.js@:
+
+```[javascript]
+  function publishMessage(channel, message) {
+    client.publish(channel, message, { qos: 1 }, function(err) {
+      if(err) {
+        console.log(err);
+      }
+    }); 
+  }
+``` 
+
+This function simply takes the message you wish to publish, and publishes it through MQTT using @client.publish@ into the channel specified. The setting 'qos' (Quality of Service) represents whether messages are guaranteed to be delivered 'at most once' (0), 'at least once' (1), or 'exactly once' (2). Here you'll use 'at least once', as you want to ensure our inputs are delivered, but also have as little back and forth communication between this device and the server as possible.
```

Comment:
arguably a game controller would be a good usecase for QoS 0: if a connection is dropped and resumed, by the time that happens the game state will probably have changed and the player'll be sending new inputs to try and direct the current state, they won't want their earlier button presses to be replayed